File: 34. Find First and Last Position of Element in Sorted Array/find_first_and_last_position_of_element_in_sorted_array.php
<?php
function firstAndLast ($element, $list){
	$findFirst = true;
	$out = [-1, -1];
	Sort($list);
	
	for ($i = 0; $i < count($list); $i++){
		$e = $list[$i];
		if ($e == $element){
			if ($findFirst){
				$out[0] = $i;
				$out[1] = $i;
				$findFirst = false;
			} else {
				$out[1] = $i;
			}
		}
	}
	
	return ($out);
}

$nums = [5,7,7,8,8,10,2];
$target = 8;

echo (json_encode(firstAndLast($target, $nums)). "\n\n\n\n");
?>

File: 34. Find First and Last Position of Element in Sorted Array/find_first_and_last_position_of_element_in_sorted_array.cpp
#include<iostream>
#include<algorithm>

using namespace std;

void first_and_last (int element, int* list, int len_list, int* out){
	bool find_first = true;
	sort (list, list + len_list);
	
	for (int i = 0; i < len_list; i++){
		if (list[i] == element){
			if (find_first){
				out[0] = i;
				out[1] = i;
				find_first = false;
			} else {
				out[1] = i;
			}
		}
	}
}

int main () {
	int nums[] = {5,7,7,8,8,10,2};
	int len_nums = sizeof(nums) / sizeof(nums[0]);
	int target = 8;
	int first_last [] = {-1,-1};
	
	first_and_last(target, nums, len_nums, first_last);
	
	cout << "This is the handy work of C++.\n\n[" << first_last[0] << ", " << first_last[1] << "]\n\n\n";
	
	return (0);
}
